File: app/domain/AnatomicalSourceType.scala
package domain

object AnatomicalSourceType extends Enumeration {
  type AnatomicalSourceType = Value
  val Blood = Value("Blood")
  val Brain = Value("Brain")
  val Colon = Value("Colon")
  val Kidney = Value("Kidney")
  val ColonAscending = Value("Ascending Colon")
  val ColonDescending = Value("Descending Colon")
  val ColonTransverse = Value("Transverse Colon")
  val Duodenum = Value("Duodenum")
  val Hair = Value("Hair")
  val Ileum = Value("Ileum")
  val Jejenum = Value("Jejenum")
  val StomachAntrum = Value("Stomach Antrum")
  val StomachBody = Value("Stomach Body")
  val Stool = Value("Stool")
  val ToeNails = Value("Toe Nails")
  val Urine = Value("Urine")
}

File: app/assets/javascripts/domain/AnatomicalSourceType/AnatomicalSourceType.js
/**
 * @author Nelson Loyola <loyola@ualberta.ca>
 * @copyright 2015 Canadian BioSample Repository (CBSR)
 */

/**
 * Anatomical sources where a specimen is collected from.
 * @enum {string}
 * @memberOf domain
 */
const AnatomicalSourceType = {
  BLOOD:            'Blood',
  BRAIN:            'Brain',
  COLON:            'Colon',
  KIDNEY:           'Kidney',
  ASCENDING_COLON:  'Ascending Colon',
  DESCENDING_COLON: 'Descending Colon',
  TRANSVERSE_COLON: 'Transverse Colon',
  DUODENUM:         'Duodenum',
  HAIR:             'Hair',
  ILEUM:            'Ileum',
  JEJENUM:          'Jejenum',
  STOMACH_ANTRUM:   'Stomach Antrum',
  STOMACH_BODY:     'Stomach Body',
  STOOL:            'Stool',
  TOE_NAILS:        'Toe Nails',
  URINE:            'Urine'
};

export default ngModule => ngModule.constant('AnatomicalSourceType', AnatomicalSourceType)
